crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -111,6 +112,65 @@ pub enum TimelineItemContent {
 }
 
 impl TimelineItemContent {
+    /// If the supplied event is suitable to be used as a latest_event in a
+    /// message preview, extract its contents and wrap it as a
+    /// TimelineItemContent.
+    #[cfg(feature = "experimental-sliding-sync")]
+    pub(crate) fn from_latest_event_content(
+        event: AnySyncTimelineEvent,
+    ) -> Option<TimelineItemContent> {
+        match is_suitable_for_latest_event(&event) {
+            PossibleLatestEvent::YesMessageLike(m) => Self::from_suitable_latest_event_content(m),
+            PossibleLatestEvent::NoUnsupportedEventType => {
+                // TODO: when we support state events in message previews, this will need change
+                warn!("Found a state event cached as latest_event! ID={}", event.event_id());
+                None
+            }
+            PossibleLatestEvent::NoUnsupportedMessageLikeType => {
+                // TODO: When we support reactions in message previews, this will need to change
+                warn!(
+                    "Found an event cached as latest_event, but I don't know how \
+                        to wrap it in a TimelineItemContent. type={}, ID={}",
+                    event.event_type().to_string(),
+                    event.event_id()
+                );
+                None
+            }
+            PossibleLatestEvent::NoEncrypted => todo!(),
```

Comment:
Should be addressed.

Reply:
Oops! Thanks for spotting that. Fixed in 6ca7693b77